### PR TITLE
Feature/project labels

### PR DESCRIPTION
**Pull request by Vitaliy-ko**

### CODE REVIEW

File: client/src/components/common/ConfirmModal/styles.module.scss [hunk 1]
```diff
@@ -0,0 +1,5 @@
+@import 'styles/mixins.scss';
+
+.primary__button {
+	@include primary-btn
+}
```
Comment (timagixe) [inline client/src/components/common/ConfirmModal/styles.module.scss:5]: We can remove this file.

File: client/src/containers/ProjectLabels/index.tsx [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+		setLabelId(labelId);
+		setIsConfirmModalOpen(true);
+	};
+
+	return (
+		<>
+			<div className={styles.body_inner__container}>
+				<div className={styles.header_inner__container}>
+					<h1 className={styles.header_inner__title}>{t('labels')}</h1>
+					<Button className={styles.primary__button} onClick={onOpenAddLabelModal}>
```
Comment (timagixe) [inline client/src/containers/ProjectLabels/index.tsx:53]:
```suggestion
					<Button className="primaryBtn" onClick={onOpenAddLabelModal}>
```

File: client/src/containers/ProjectLabels/index.tsx [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+									isFullSize={true}
+									backgroundColor={label.backgroundColor}
+									text={label.text}
+									textColor={label.textColor}
+								/>
+								<span className={styles.labels__list_actions}>
+									<Button
+										basic
+										secondary
+										className={styles.primary__button_basic}
```
Comment (timagixe) [inline client/src/containers/ProjectLabels/index.tsx:78]:
```suggestion
										className='contentBtn'
```

File: client/src/containers/ProjectLabels/index.tsx [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+										onClick={() => onOpenEditLabelModal(label)}
+									>
+										{t('edit')}
+									</Button>
+									<Button
+										color="red"
```
Comment (timagixe) [inline client/src/containers/ProjectLabels/index.tsx:84]:
```suggestion
										className="cancelBtn"
```

File: client/src/containers/ProjectLabels/styles.module.scss [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+		}
+	}
+}
+.label__example_title {
+	margin-right: 5px;
+}
+
+.primary__button {
+	@include primary-btn
+}
```
Comment (timagixe) [inline client/src/containers/ProjectLabels/styles.module.scss:74]: Let's remove it and use className="primaryBtn" instead.

File: client/src/containers/ProjectLabels/styles.module.scss [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+.labels__list_item {
+	display: flex;
+	justify-content: space-between;
+	align-items: center;
+	margin-bottom: 10px;
+}
+
+.labels__list_actions {
+	display: flex;
+}
```
Comment (timagixe) [inline client/src/containers/ProjectLabels/styles.module.scss:93]: New line at the end of file.

File: client/src/containers/Projects/table.tsx [hunk 7]
```diff
@@ -104,12 +106,13 @@ const ProjectsTable = ({ projects, currentUser }: Props) => {
 							</Table.Cell>
 							<Table.Cell>{key}</Table.Cell>
 							<Table.Cell>Software</Table.Cell>
-							<Table.Cell className={styles.project__lead_container}>
-								<UserAvatar user={lead} small />
-								<Link to={`/profile/${lead.id}`} children={`${lead.firstName} ${lead.lastName}`} />
+							<Table.Cell className={styles.project__lead_wrapper}>
+								<span className={styles.project__lead_container}>
+									<UserAvatar user={lead} small />
+									<span>{`${lead.firstName} ${lead.lastName}`}</span>
```
Comment (timagixe) [inline client/src/containers/Projects/table.tsx:112]:
```suggestion
									<span>{getUsername(lead)}</span>
```

File: client/src/services/projects.service.ts [hunk 8]
```diff
@@ -87,3 +88,48 @@ export const updateProjectUsersList = async ({ usersId, projectId }: actionTypes
...
+	return (await res.json()) as WebApi.Entities.ProjectLabel;
+};
+
+export const deleteLabel = async ({
+	projectId,
+	labelId,
+}: {
+	projectId: string;
+	labelId: string;
+}): Promise<WebApi.Entities.ProjectLabel> => {
```
Comment (timagixe) [inline client/src/services/projects.service.ts:124]:
```suggestion
export const deleteLabel = async ({ projectId, labelId }: DeleteLabel): Promise<WebApi.Entities.ProjectLabel> => {
```

File: client/src/services/projects.service.ts [hunk 9]
```diff
@@ -1,3 +1,4 @@
+import { Label } from 'containers/ProjectLabels/logic/actionTypes';
```
Comment (timagixe) [inline client/src/services/projects.service.ts:1]:
```suggestion
import { Label, DeleteLabel } from 'containers/ProjectLabels/logic/actionTypes';
```

File: server/src/controllers/projects.controllers.ts [hunk 10]
```diff
@@ -253,6 +255,130 @@ class ProjectsController {
...
+			const isLabelDataValid = this.validateLabelData(label, next);
+			if (!isLabelDataValid) {
+				return;
+			}
+
+			const project = await this.getProjectById(projectId, userId, projectsRepository, next);
+
+			if (!project) {
+				return;
+			}
```
Comment (timagixe) [inline server/src/controllers/projects.controllers.ts:274]: Do we really need this?

From what I see `getProjectById` itself will throw an error if it cannot find project
```
		if (project === undefined) {
			next(new ErrorResponse(HttpStatusCode.NOT_FOUND, projectsErrorMessages.PROJECT_NOT_FOUND));
			return null;
		}
```

If we can remove it here then it should be removed from `updateLabel` and `deleteLabel` too.

**Review by timagixe — CHANGES_REQUESTED** [review]

File: client/src/components/common/ConfirmModal/styles.module.scss [hunk 11]
```diff
@@ -0,0 +1,5 @@
+@import 'styles/mixins.scss';
+
+.primary__button {
+	@include primary-button
+}
```
[the remark this answers is not in the document]
Comment (Vitaliy-ko) [reply client/src/components/common/ConfirmModal/styles.module.scss]: Removed

File: client/src/containers/ProjectLabels/AddLabelModal.tsx [hunk 12]
```diff
@@ -0,0 +1,142 @@
...
+						<Form.Field>
+							<label>
+								<span className={styles.label__example_title}>{t('result')}</span>
+								<Label backgroundColor={backgroundColor} text={text} textColor={textColor} />
+							</label>
+						</Form.Field>
+					</Form>
+				</Modal.Content>
+				<Modal.Actions>
+					<Button onClick={onModalClose}>{t('cancel')}</Button>
```
[the remark this answers is not in the document]
Comment (Vitaliy-ko) [reply client/src/containers/ProjectLabels/AddLabelModal.tsx:124]: Added class cancelBtn

File: client/src/containers/ProjectLabels/AddLabelModal.tsx [hunk 13]
```diff
@@ -0,0 +1,142 @@
...
+					<>
+						{!isEditMode ? (
+							<Button className={styles.primary__button} onClick={onAddLabel} loading={isLoading}>
+								{t('add')}
+							</Button>
+						) : (
+							<Button className={styles.primary__button} onClick={onEditLabel} loading={isLoading}>
```
[the remark this answers is not in the document]
Comment (Vitaliy-ko) [reply client/src/containers/ProjectLabels/AddLabelModal.tsx:131]: Added primaryBtn class

File: client/src/components/common/ConfirmModal/index.tsx [hunk 14]
```diff
@@ -17,7 +18,7 @@ const ConfirmModal = ({ isOpened, setIsOpened, confirmAction, header, content }:
 		</Modal.Content>
 		<Modal.Actions>
 			<Button onClick={() => setIsOpened(false)}>No</Button>
-			<Button primary onClick={confirmAction}>
+			<Button primary onClick={confirmAction} className={styles.primary__button}>
```
[the remark this answers is not in the document]
Comment (Vitaliy-ko) [reply client/src/components/common/ConfirmModal/index.tsx:21]: Added suggested changes

File: client/src/components/common/Label/styles.module.scss [hunk 15]
```diff
@@ -0,0 +1,13 @@
...
+	margin-right: 5px;
+	font-size: .92857143em;
+    font-weight: 700;
+	text-transform: none;
+}
+
+.full_size {
+	font-size: 1.3em;
+	padding: 7px 10px;
+}
```
[the remark this answers is not in the document]
Comment (Vitaliy-ko) [reply client/src/components/common/Label/styles.module.scss:13]: Added new line at the end of the file

File: client/src/containers/ProjectLabels/styles.module.scss (hunk 6 above)
Comment (Vitaliy-ko) [reply client/src/containers/ProjectLabels/styles.module.scss:93]: Added new line

File: client/src/containers/Projects/table.tsx (hunk 7 above)
Comment (Vitaliy-ko) [reply client/src/containers/Projects/table.tsx:112]: Added suggested changes

File: client/src/services/projects.service.ts (hunk 8 above)
Comment (Vitaliy-ko) [reply client/src/services/projects.service.ts:124]: Added suggested changes

File: client/src/services/projects.service.ts (hunk 9 above)
Comment (Vitaliy-ko) [reply client/src/services/projects.service.ts:1]: Added suggested changes

File: client/src/containers/ProjectLabels/AddLabelModal.tsx [hunk 16]
```diff
@@ -0,0 +1,142 @@
...
+			setIsValidErrorShown(true);
+			return;
+		}
+
+		dispatch(
+			actions.startAddingLabel({
+				project,
+				label: { text, textColor, backgroundColor },
+			}),
+		);
```
[the remark this answers is not in the document]
Comment (Vitaliy-ko) [reply client/src/containers/ProjectLabels/AddLabelModal.tsx:71]: You are right I've missed it...((

**Pull request merged**